code/modules/black_market/black_market_item.dm

```diff
@@ -204,18 +196,82 @@ var/locations_calculated = 0
 			A.put_in_any_hand_if_possible(spawned_item)
 
 	log_transaction("The item was teleported directly to him.", user)
-	radio.visible_message("The [radio.name] beeps: <span class='warning'>Thank you for your purchase!</span>")
+	radio.visible_message("\The [radio] beeps: <span class='warning'>Thank you for your purchase!</span>")
 	radio.interact(user)
 
-	spawn(rand(30 SECONDS, 60 SECONDS))
-		if(!radio.nanotrasen_variant && prob(sps_chances[EXPENSIVE]))
-			SPS_black_market_alert("Centcomm has detected a black market purchase of item: [name]. It was teleported directly to the buyer in the past minute.")
-
-
 /datum/black_market_item/proc/after_spawn(var/obj/spawned, var/delivery_method, var/mob/user) //Called immediately after spawning. Override for post-spawn behavior.
 	return
 
+/*
+//
+//	PLAYER MARKET DATUM
+//
+*/	
+	
+var/list/player_market_items = list()
+
+/datum/black_market_player_item
+	var/atom/item
+	var/obj/item/device/black_market_beacon/attached_beacon
+	var/obj/item/device/illegalradio/seller_radio
+	var/mob/living/seller
```

Comment:
All of these should be set to null on Destroy.
Alternatively, rather than storing a hard reference to a mob, store a ckey or a mind.
The reason is that this ref will be held and prevent the `seller` from being GC'd.